framework/base/Behavior.php

```diff
@@ -85,7 +90,7 @@ public function attach($owner)
     public function detach()
     {
         if ($this->owner) {
-            foreach ($this->events() as $event => $handler) {
+            foreach ($this->_attachedEvents as $event => $handler) {
```

Comment:
Maybe we should call `unset($this->_attachedEvents[$event]);` after `$this->owner->off()`? Or reset `$_attachedEvents` at the and of `detach()`?

Reply:
Done.